internal/resource/common/pluginconfiguration/plugin_configuration_state.go

```diff
@@ -131,7 +133,7 @@ func ToRowsListValue(rows []client.ConfigRow, planRows types.List, diags *diag.D
 	return listVal
 }
 
-func ToTablesListValue(tables []client.ConfigTable, planTables types.List, diags *diag.Diagnostics) types.List {
+func ToTablesListValue(tables []client.ConfigTable, planTables types.List, diags *diag.Diagnostics) (types.List, diag.Diagnostics) {
```

Comment:
These methods don't need to return diagnostics, they are adding the the pointer passed in